wnfs/src/private/file.rs

```diff
@@ -62,29 +108,370 @@ impl PrivateFile {
+    ///
+    ///     assert_eq!(content, all_content);
+    /// }
+    /// ```
+    pub async fn get_content<B: BlockStore>(
+        &self,
+        hamt: &PrivateForest,
+        store: &B,
+    ) -> Result<Vec<u8>> {
+        let mut content = Vec::with_capacity(self.get_content_size_upper_bound());
+        let mut stream = self.stream_content(hamt, store);
+        while let Some(bytes) = stream.next().await {
+            content.extend_from_slice(&bytes?);
+        }
+        Ok(content)
+    }
+
+    /// Gets the block at specified index.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::rc::Rc;
+    /// use chrono::Utc;
+    /// use rand::thread_rng;
+    /// use wnfs::{
+    ///     private::{PrivateForest, PrivateRef},
+    ///     MemoryBlockStore, Namefilter, PrivateFile,
+    ///     utils::get_random_bytes, MAX_BLOCK_SIZE
+    /// };
+    ///
+    /// #[async_std::main]
+    /// async fn main() {
+    ///     let store = &mut MemoryBlockStore::default();
+    ///     let rng = &mut thread_rng();
+    ///     let hamt = Rc::new(PrivateForest::new());
+    ///
+    ///     let content = get_random_bytes::<100>(rng).to_vec();
+    ///     let (file, hamt) = PrivateFile::with_content(
+    ///         Namefilter::default(),
+    ///         Utc::now(),
+    ///         content.clone(),
+    ///         hamt,
+    ///         store,
+    ///         rng,
+    ///     )
+    ///     .await
+    ///     .unwrap();
+    ///
+    ///     let mut content_at_0 = file.get_block_at(0, &hamt, store).await.unwrap();
+    ///
+    ///     assert_eq!(content, content_at_0.unwrap());
+    /// }
+    /// ```
+    pub async fn get_block_at<B: BlockStore>(
+        &self,
+        index: usize,
+        hamt: &Rc<PrivateForest>,
+        store: &B,
+    ) -> Result<Option<Vec<u8>>> {
+        match &self.content {
+            FileContent::Inline { data } => {
+                if index == 0 {
+                    return Ok(Some(data.clone()));
+                }
+
+                Ok(None)
+            }
+            FileContent::External {
+                key, block_count, ..
+            } => {
+                if index < *block_count {
+                    let label = Self::create_block_label(key, *block_count, &self.header.bare_name);
+                    let bytes = Self::decrypt_block(key, &label, hamt, store).await?;
+                    return Ok(Some(bytes));
+                }
+
+                Ok(None)
+            }
+        }
+    }
+
+    /// Determines where to put the content of a file. This can either be inline or stored up in chunks in a HAMT.
+    pub(super) async fn prepare_content<B: BlockStore, R: RngCore>(
+        bare_name: &Namefilter,
+        content: Vec<u8>,
+        mut hamt: Rc<PrivateForest>,
+        store: &mut B,
+        rng: &mut R,
+    ) -> Result<(FileContent, Rc<PrivateForest>)> {
+        // TODO(appcypher): Use a better heuristic to determine when to use external storage.
```

Comment:
Still a remaining TODO?

Perhaps its just safer to always go external content by default?

I'm fine with leaving this here, just wanted to raise this in case it was accidental. I think it's unlikely that metadata on a node exceeds ~50kb.